librb/src/linebuf.c

```diff
@@ -545,13 +545,12 @@ rb_linebuf_flush(rb_fde_t *F, buf_head_t * bufhead)
  * autoconf checks for this..but really just want to use it if we have a
  * native version even if libircd provides a fake version...
  */
-#ifdef HAVE_WRITEV
 	if(!rb_fd_ssl(F))
 	{
 		rb_dlink_node *ptr;
 		int x = 0, y;
 		int xret;
-		static struct rb_iovec vec[RB_UIO_MAXIOV];
+		static struct rb_iovec vec[IOV_MAX];
```

Comment:
This is a step backwards in some way, since POSIX says that the `IOV_MAX` constant may not be available (in which case the application has to call `getconf(_SC_IOV_MAX)` if the guaranteed minimum `_XOPEN_IOV_MAX` is insufficient). On the other hand, not using `UIO_MAXIOV` seems good.

I think the best approach would be to call `getconf(_SC_IOV_MAX)` (once at startup) and apply some reasonable maximum such as 1024 to it. The array can still be static in this case. If `getconf(_SC_IOV_MAX)` returns -1, the value `_XOPEN_IOV_MAX` can be used. I believe using `_XOPEN_IOV_MAX` unconditionally will lead to unnecessary extra system calls, since it's only 16, it's common to send more lines than that at once and practical systems allow much more.

It may make more sense to move this part to a separate pull request.

Reply:
Done! Thanks!